core/src/main/java/org/apache/struts2/interceptor/coop/CoopInterceptor.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.interceptor.coop;
+
+import com.opensymphony.xwork2.ActionInvocation;
+import com.opensymphony.xwork2.interceptor.AbstractInterceptor;
+import com.opensymphony.xwork2.interceptor.PreResultListener;
+import com.opensymphony.xwork2.util.TextParseUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CoopInterceptor extends AbstractInterceptor implements PreResultListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CoopInterceptor.class);
+
+    private CoopConfiguration config = new CoopConfiguration();
+
+    public void setDefaultMode(String mode){
+        config.setDefaultMode(mode);
+    }
+
+    public void setExemptedPaths(String paths){
+        config.setExemptedPaths(TextParseUtil.commaDelimitedStringToSet(paths));
+    }
+
+    @Override
+    public String intercept(ActionInvocation invocation) throws Exception {
+        invocation.addPreResultListener(this);
+        return invocation.invoke();
+    }
+
+    @Override
+    public void beforeResult(ActionInvocation invocation, String resultCode) {
+        HttpServletRequest request = invocation.getInvocationContext().getServletRequest();
+        HttpServletResponse response = invocation.getInvocationContext().getServletResponse();
+        String path = request.getContextPath();
+
+        if (config.isExempted(path)){
+            // no need to add headers
+            LOG.debug("No need to add COOP header, path is exempted");
```

Comment:
Let's say something along the lines of 'Skipping COOP header for exempted path X'